numba/tests/test_record_dtype.py

```diff
@@ -663,6 +663,37 @@ def test_record_two_arrays(self):
         cfunc = self.get_cfunc(pyfunc, (nbrecord,))
         self.assertEqual(cfunc(rec), pyfunc(rec))
 
+    def test_structure_dtype_with_titles(self):
+        # the following is the definition of int4 vector type from pyopencl
+        vecint4 = np.dtype([(('x', 's0'), '<i4'), (('y', 's1'), '<i4'),
```

Comment:
Just curious: `<i4` won't work a big-endian machine since we only support native-endian arrays, right?